### DISCUSSION TITLE
getServerSideProps is requesting an unknown request. Is this a bug?

### DISCUSSION THREAD

**Discussion by Polgasol** [vercel/next.js#40755] [Help]
Everytime I make an api request, an unknown request by getserversideprops is being made. I'm using React-query at the same time. My code:
 
`export const getServerSideProps: GetServerSideProps = async (ctx) => {`
  `try {`
  `   const queryClient = new QueryClient()`
     `await queryClient.fetchQuery(['userAuth'], () => userAuth(ctx.req))`

    return {
      props: {
        dehydratedState: dehydrate(queryClient),
      },
    };
  } catch (err) {

    return {
      props: {
        error: {
          message: `Internal Error`,
          statusCode: 404,
        },
      },
   ` };`
 ` }`
`};`
My server terminal always show this:
`{`
  `accept: 'application/json, text/plain, */*',`
 ` 'user-agent': 'axios/0.27.2',`
  `host: 'localhost:5000',`
  `connection: 'close'`
`}`
If I remove getserversideprops and just request in client side, I'm not getting this api call. Is this a bug?

**Comment by icyJoseph** [comment]
What's `userAuth` doing?

**Reply by Polgasol** [reply]
Here is my userAuth api:
`import axios from 'axios';`

`export const userAuth = async (req: any) => {`
`if (req.headers.cookie === undefined) {`
`const res: any = await axios.get('http://localhost:5000/api', {`
`withCredentials: true`
`});`
`return res.data.data;`
`}`
`const res: any = await axios.get('http://localhost:5000/api', {`
`headers: {`
`Cookie: req.headers.cookie,`
`},`
`withCredentials: true`
`});`
`return res.data.data;`
`}`

**Reply by Polgasol** [reply]
Literally if I remove getServersideProps, nextjs is not calling this request. I'm using cors and the origin is undefined as well.

**Reply by Polgasol** [reply]
Cors file:
`import cors from 'cors';`
`import logger from '../../logger/index';`

`const whitelist: Set<string | undefined> = new Set([`
 ` 'http://localhost:3000', `
`]);`

`const corsOptions: cors.CorsOptions = {`
 ` optionsSuccessStatus: 200,`
 ` origin: (origin, callback) => {`
  `  if (whitelist.has(origin) || !origin) {`
  `    callback(null, true);`
`    } else {`
    `  callback(new Error('May error'), false);`
 `   }`
`  },`
`  credentials: true, `
`};`

middlewares:
`server.options('*', cors);` 
  `server.use(cors);`
 `server.set('trust proxy', true);`
  `server.use(cookieParser());`
  `server.use(bodyParser.urlencoded({ extended: true }));`
 ` server.use(express.urlencoded({ extended: true }));`
 ` server.use(express.json());`
`server.use(sessions);`
  `server.use(passport.initialize());`
  `server.use(passport.session());`

**Reply by icyJoseph** [reply]
Let me first get one thing out. That is [getServerSideProps or API Routes](https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props#getserversideprops-or-api-routes), which are in a sense one of the same. A good mental model is to think of the `getServerSideProps` function as a specialised API route, whose result is directly passed as props to the Page Component. It is not 100% truth, but it helps when starting out.

So why would we bounce a request from an specialised API route, to a generic API route, only to forward that once again to the service we want to query/mutate? It is an unnecessary jump!

Keep in mind that `getServerSideProps`(GSSP) is strictly scoped to the server. It's code and dependencies are never bundled to the client, use this tool to convince yourself https://next-code-elimination.vercel.app/, and it never runs on the client. You can directly pool a database, or any other services you need to from `GSSP`.

In your case, it does make sense that origin is `null` because the request is originating from within the server. It is really the server reaching to itself.

As we unravel this mystery I'll eventually EDIT the top comment and replace it with what we come to conclude. 

So my first guess would be that your `logger` is picking up on the additional jump made from `GSSP` to `/api`?

--- 

By the way, you can past block codes with \`\`\` for example a bit of javascript would start with \`\`\`js, then press enter and write some JS, then in a new line \`\`\` again

> \`\`\`js
>  const noop = () => {}
> \`\`\`
